### PR TITLE
Navigation block: e2e code quality fixes

### PR DESCRIPTION
**Pull request by MaggieCabrera**
## What?

Followup on https://github.com/WordPress/gutenberg/pull/47867 after I missed a couple of things that were found on review after merge.

This PR also includes the `no-page-pause` rule to the linter so people like me won't commit and merge pauses to their e2e tests :D

To test check that the navigation block tests pass (`npm run test:e2e:playwright -- editor/blocks/navigation.spec.js`) and that you can't commit a test containing `page.pause()`.

### CODE REVIEW

**Comment by MaggieCabrera** [comment]
@kevin940726 do you know what else needs to be done to add the `no-page-pause` rule? I'm getting 

`error  Definition for rule 'no-page-pause' was not found  no-page-pause`

File: .eslintrc.js [hunk 1]
```diff
@@ -392,6 +392,7 @@ module.exports = {
 			rules: {
 				'@wordpress/no-global-active-element': 'off',
 				'@wordpress/no-global-get-selection': 'off',
+				'no-page-pause': 'error',
```
Comment (kevin940726) [inline .eslintrc.js]: Let's try namespacing it.

```suggestion
				'playwright/no-page-pause': 'error',
```

File: packages/e2e-test-utils-playwright/src/request-utils/menus.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export interface NavigationMenu {
 /**
  * Create a classic menu
  *
- * @param {string} name Menu name.
+ * @param  name Menu name.
  * @return {string} Menu content.
```
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/request-utils/menus.ts]: Nit: We also don't need the return type either.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 3]
```diff
@@ -78,7 +71,11 @@ test.describe(
 			await editor.insertBlock( { name: 'core/navigation' } );
 
 			// Check the block in the canvas.
-			await navBlockUtils.selectNavigationItemOnCanvas( 'WordPress' );
+			await expect(
+				this.editor.canvas.locator(
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: This should just be `editor`.

```suggestion
				editor.canvas.locator(
```

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 4]
```diff
@@ -91,14 +88,18 @@ test.describe(
 			await page.locator( 'role=button[name="Close panel"i]' ).click();
 
 			// Check the block in the frontend.
-			await navBlockUtils.selectNavigationItemOnFrontend( 'WordPress' );
+			await this.page.goto( '/' );
+			await expect(
+				this.page.locator(
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Same here:

```suggestion
			await page.goto( '/' );
			await expect(
				page.locator(
```

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 5]
```diff
@@ -113,42 +114,23 @@ test.describe(
 			] );
 
 			// Check the block in the canvas.
-			await editor.page.pause();
-			await navBlockUtils.selectNavigationItemOnCanvas( 'Custom link' );
+			await expect(
+				this.editor.canvas.locator(
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Ditto.

```suggestion
				editor.canvas.locator(
```

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 6]
```diff
@@ -113,42 +114,23 @@ test.describe(
...
+					`role=textbox[name="Navigation link text"i] >> text="Custom link"`
+				)
+			).toBeVisible();
 
 			// Check the block in the frontend.
-			await navBlockUtils.selectNavigationItemOnFrontend( 'Custom link' );
-			await editor.page.pause();
+			await this.page.goto( '/' );
+			await expect(
+				this.page.locator(
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Ditto.

```suggestion
			await page.goto( '/' );

			await expect(
				page.locator(
```

File: .eslintrc.js (hunk 1 above)
Comment (MaggieCabrera) [reply .eslintrc.js]: mmh, that still allows the pauses on the tests, I'm not getting any errors when I try to commit them

Comment (kevin940726) [reply .eslintrc.js]: Hmm, it's working for me though. 🤔

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 7]
```diff
@@ -91,14 +88,18 @@ test.describe(
...
+			await page.goto( '/' );
+			await expect(
+				page.locator(
+					`role=navigation >> role=link[name="WordPress"i]`
+				)
+			).toBeVisible();
 		} );
 
 		test( 'default to the only existing classic menu if there are no block menus', async ( {
 			admin,
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:100]: We'll need to add `page` here to use it inside the test.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 8]
```diff
@@ -113,42 +115,24 @@ test.describe(
 			] );
 
 			// Check the block in the canvas.
-			await editor.page.pause();
-			await navBlockUtils.selectNavigationItemOnCanvas( 'Custom link' );
+			await expect(
+				editor.canvas.locator(
+					`role=textbox[name="Navigation link text"i] >> text="Custom link"`
+				)
+			).toBeVisible();
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:122]: Interesting that this is already reported as flaky (https://github.com/WordPress/gutenberg/issues/48033). We might want to take a look at this test 🙂 .

**Review by kevin940726 — APPROVED** [review]
LGTM 👍 Thanks!

Note that there's already a flaky test report in https://github.com/WordPress/gutenberg/issues/48033. We might want to take a look at that in another PR 🙂.

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 8 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js:122]: you are right, I'll have a look tomorrow!

**Pull request merged**